libexec/rtld-elf/rtld_c18n.c

```diff
@@ -1429,6 +1439,7 @@ _rtld_sigaction_begin(int sig, struct sigaction *act)
 		    "a signal handler");
 
 	act->sa_flags |= SA_SIGINFO;
+	act->sa_flags &= ~SA_ONSTACK;
```

Comment:
Yeah you can't just do this. If a compartment asks for signal delivery on a separate stack, you need to respect that. Maybe for *now* it's fine to do, and likely the software we're running doesn't care, but there is real-world software out there that absolutely needs this to be adhered to, and so this needs at least an XXX stating that this is wrong and broken.